plugins/flytekit-omegaconf/flytekitplugins/omegaconf/dictconfig_transformer.py

```diff
@@ -0,0 +1,178 @@
+import importlib
+import logging
+import re
+import typing
+from typing import Type, TypeVar
+
+import flatten_dict
+from flyteidl.core.literals_pb2 import Literal as PB_Literal
+from flytekitplugins.omegaconf.config import OmegaConfTransformerMode, SharedConfig
+from flytekitplugins.omegaconf.type_information import extract_node_type
+from google.protobuf.json_format import MessageToDict, ParseDict
+from google.protobuf.struct_pb2 import Struct
+
+import omegaconf
+from flytekit import FlyteContext
+from flytekit.core.type_engine import TypeTransformerFailedError
+from flytekit.extend import TypeEngine, TypeTransformer
+from flytekit.models.literals import Literal, Scalar
+from flytekit.models.types import LiteralType, SimpleType
+from omegaconf import DictConfig, OmegaConf
+
+logger = logging.getLogger("flytekit")
+
+T = TypeVar("T")
+NoneType = type(None)
+
+
+def is_flattenable(d: DictConfig) -> bool:
+    """Check if a DictConfig can be properly flattened and unflattened, i.e. keys do not contain dots."""
+    return all(
+        isinstance(k, str)  # keys are strings ...
+        and "." not in k  # ... and do not contain dots
+        and (
+            OmegaConf.is_missing(d, k)  # values are either MISSING ...
+            or not isinstance(d[k], DictConfig)  # ... not nested Dictionaries ...
+            or is_flattenable(d[k])
+        )  # or flatable themselves
+        for k in d.keys()
+    )
+
+
+class DictConfigTransformer(TypeTransformer[DictConfig]):
+    def __init__(self, mode: typing.Optional[OmegaConfTransformerMode] = None):
```

Comment:
As discussed [here](https://github.com/flyteorg/flytekit/pull/2299/files#r1660227529), shell we remove the `mode` here and only provide a global mode?

Reply:
Addressed in https://github.com/flyteorg/flytekit/pull/2299/commits/e6e18964ff9d0f53a6722d1320e239f3d3a96545

We provide global mode and a context manager for ability to control it locally still.